### DISCUSSION TITLE
Linkerd2.1.1 init error

### DISCUSSION THREAD

**Discussion by zhanglt** [linkerd/linkerd2#7576] [Q&A]
centos8.5.2111  kernle:5.15.13-1.el8.elrepo.x86_64
linkerd2.11.1
[root@master ~]# k get pod
NAME                                      READY   STATUS                  RESTARTS        AGE
linkerd-destination-754fdff96f-tzcf6      0/4     Init:CrashLoopBackOff   7 (2m18s ago)   14m
linkerd-identity-54795b9f9f-pq7lh         0/2     Init:CrashLoopBackOff   7 (2m25s ago)   14m
linkerd-proxy-injector-69c975fc7c-fvt4l   0/2     Init:CrashLoopBackOff   7 (77s ago)     14m

 k describe pod linkerd-identity-54795b9f9f-pq7lh

2022/01/09 11:55:41 adding rules
------------------------------------------------------------
2022/01/09 11:55:41 :; iptables -t nat -N PROXY_INIT_REDIRECT -m comment --comment proxy-init/redirect-common-chain/1641729341
2022/01/09 11:55:41 modprobe: can't change directory to '/lib/modules': No such file or directory
modprobe: can't change directory to '/lib/modules': No such file or directory
iptables v1.8.7 (legacy): can't initialize iptables table `nat': Table does not exist (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.

2022/01/09 11:55:41 Aborting firewall configuration
Error: exit status 3

**Comment by alpeb** [comment]
Can you describe a bit more your environment? (kubernetes version, cluster environment (GKE, AKS, kops, ...), how you installed linkerd (CLI, helm, cni, ...))

**Reply by cccsss01** [reply]
I am having the same issue,  Kubeadm version 1.23.1, bare metal, installed with CLI,  received warnings @ install:
warning: spec.template.spec.nodeSelector[beta.kuberernetes.io/os]: deprecated since v1.14, use kubernetes.io/os, same with cronjob.
I'm on ALmaLinux 4.18.0-348.7.1.el8_5.x86_64 gcc version 8.5.0 20210514 (RedHat 8.5.0-4)

calico for CNI, i'm thinking this might be related to nf_tables?
